src/Kernels/Receiver.cpp

```diff
@@ -82,8 +83,20 @@ double seissol::kernels::ReceiverCluster::calcReceivers(  double time,
   alignas(ALIGNMENT) real timeEvaluatedAtPoint[tensor::QAtPoint::size()];
   alignas(ALIGNMENT) real timeEvaluatedDerivativesAtPoint[tensor::QDerivativeAtPoint::size()];
 #ifdef ACL_DEVICE
-  constexpr size_t dofsHostCopySize = tensor::Q::size();
-  alignas(ALIGNMENT) real dofsHostCopy[ dofsHostCopySize ];
+  if (deviceCollector.get() == nullptr) {
+    deviceIndices.resize(m_receivers.size());
+    std::vector<real*> dofs;
+    std::unordered_map<real*, size_t> indexMap;
```

Comment:
I don't understand what is indexMap.
I mean:
`indexMap[currentDofs] = dofs.size();`
does not really suggest `indexMap[currentDofs]` points to an index but to a size?

Reply:
We add an element directly after that—i.e. `currentDofs` is then located at `indexMap[currentDofs]`.